test/utils/SafeCast.test.js

```diff
@@ -43,4 +43,74 @@ describe('SafeCast', async () => {
   }
 
   [8, 16, 32, 64, 128].forEach(bits => testToUint(bits));
+
+  describe('toUint256', () => {
+    const maxInt256 = new BN('2').pow(new BN(255)).subn(1);
+    const minInt256 = new BN('2').pow(new BN(255));
```

Comment:
```suggestion
    const minInt256 = new BN('2').pow(new BN(255)).neg();
```

It's sort of strange that tests are passing given this, but I guess it has to do with how web3 encodes arguments. If this is interpreted as a 32 byte value, then 2^255 would correspond to -1.

Reply:
Using two's complement representation for signed integers of 32 bytes, 2^255 corresponds to -2^255.
Even if the BN in the test would use more, the moment the number is fed to the Solidity function, it gets truncated to 32 bytes (this can be easily checked by feeding it 2^256, that will be successfully cast to 0).
 
I think this is the reason tests are passing with and without the negation.

Also, I fully agree explicit is better, I will add the negation; as before, I was thinking in bits :)